src/ds5/ds5-device.cpp

```diff
@@ -670,38 +681,72 @@ namespace librealsense
         _hw_monitor->get_gvd(gvd_buf.size(), gvd_buf.data(), GVD);
 
         // Opaque retrieval
-        d400_caps val{d400_caps::CAP_UNDEFINED};
-        if (gvd_buf[active_projector])  // DepthActiveMode
-            val |= d400_caps::CAP_ACTIVE_PROJECTOR;
-        if (gvd_buf[rgb_sensor])                           // WithRGB
-            val |= d400_caps::CAP_RGB_SENSOR;
-        if (gvd_buf[imu_sensor])
+        d400_caps val{ d400_caps::CAP_UNDEFINED };
+
+        if (!(_pid == RS_D585_PID || _pid == RS_S585_PID))
```

Comment:
Maybe split to functions? it's a large if/else
This way we will only modify the new one each update for safety camera